tinygrad/tensor.py

```diff
@@ -3183,7 +3183,8 @@ def cast(self, dtype:DTypeLike) -> Tensor:
     print(t.dtype, t.numpy())
     ```
     """
-    return self if self.dtype == (dt:=to_dtype(dtype)) else F.Cast.apply(self, dtype=dt)
+    if (dt := to_dtype(dtype)) == dtypes.uint8 and self.dtype == dtypes.float: return F.Cast.apply(F.Cast.apply(self, dtype=dtypes.int64), dtype=dt)
```

Comment:
i think casting twice to make it correct is fine. but only this pair is too specific. can you check if all float to all uint works, in particular some uint64 should fail because int64 cannot fully represent it